publish-subscribe/src/main/java/com/iluwatar/publish/subscribe/model/Topic.java

```diff
@@ -0,0 +1,59 @@
+package com.iluwatar.publish.subscribe.model;
+
+import com.iluwatar.publish.subscribe.subscriber.Subscriber;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+
+/** This class represents a Topic that topic name and subscribers. */
+public class Topic {
+
+  private final TopicName name;
+  private final Set<Subscriber> subscribers = new CopyOnWriteArraySet<>();
+
+  /**
+   * Creates a new instance of the Topic class.
+   *
+   * @param name The name of the topic.
+   */
+  public Topic(TopicName name) {
+    this.name = name;
+  }
+
+  /**
+   * Get the name of the topic.
+   *
+   * @return topic name
+   */
+  public TopicName getName() {
+    return name;
+  }
```

Comment:
Use Lombok annotations to get rid of boilerplate, such as simple constructors, getter, setters, etc.